test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -583,8 +613,27 @@ private NodeAndClient buildNode(int nodeId, long seed, Settings settings, Versio
         return new NodeAndClient(name, node);
     }
 
-    private String buildNodeName(int id) {
-        return nodePrefix + id;
+    private String buildNodeName(int id, Settings settings) {
+        String prefix = nodePrefix;
+        prefix = prefix + getRoleSuffix(settings);
+        return prefix + id;
```

Comment:
I'm not sure, this goes in each logs? we now have node_sm1 or node_sd1 . If we change it will be `node_s_m_1` and `node_s_d_1` . Feels like an overkill?